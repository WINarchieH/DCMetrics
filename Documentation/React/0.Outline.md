## React Documentation

The React documentation discusses 

### Table of Content

- [React: General design and page structure](1.React.md)
- [Redux: Persisting session information](2.Redux.md)
- [React-Table: Displaying table information](3.React-Table.md)
- [Hooks: Useful React hooks](4.Hooks.md)
- [Routing: Ensuring secure page access](5.Routing.md)


### Extensions

### Screen Change Tab Name 

- Use `react-table` to change title for each screen.