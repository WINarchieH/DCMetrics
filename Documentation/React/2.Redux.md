# 2. Redux Documentation 

[Return to React Outline](0.Outline.md)

Redux is used for 

- Storing session information 
- Maintaining session information across screens

Redux **does not** 

- Maintain session information across tabs 

**Note.** Session information across tabs are achieved using session storage on **Express** and 
is retrieved in [PrivateRoutes](../../React/src/components/routes/privateRoute.js).
More information can be found in [Routing](5.Routing.md#PrivateRoute).

## Table of Contents

- [General Structure](#General-Structure)
- [Storing Session Information](#Storing-Session-Information)
- [Retrieving Session Information](#Retrieving-Session-Information)
- [Other Information](#Other-Information)
- [Useful Tools](#Useful-Tools)

## General Structure

The workflow for adding a variable/action to Redux is the following:

- Define a **variable name** and an **action** that is applied to the variable in [actions](../../React/src/redux/actions.js).
- Specify the **action** name in [actionTypes](../../React/src/redux/actionTypes.js). (This is mainly so that all the defined actions can be quickly found.)
- Define how the **action** works and possible **default values** for the variable involved in a new `.js` file in the [reducers directory](../../React/src/redux/reducers/). 
This will be a function and is called a *reducer*.
- Specify the newly defined reducer in the [index reducer file](../../React/src/redux/reducers/index.js). 
This exposes the action so that it can be used in any screen to retrieve session information.

### Example 

```javascript
    // actionTypes.js 
    export const ACTION_NAME = 'ACTION_NAME';
    // actions.js 
    import { ACTION_NAME, ... } from './actionTypes';
    /// Passing a value
    export const actionName = variable => ({
        type: ACTION_NAME,
        body: { variable }
    });
    // ./reducers/variable.js 
    import { ACTION_NAME } from './actionTypes';
    const initialState = INITIAL_VALUES;

    export const variable = (state=initialState, action) => {
        switch (action.type) {
            case ACTION_NAME: {
                return action.body.variable;
            }
            default: {
                return state;
            }
        }
    };
```

In this example, `variable` is stored in Redux and the *action* associated 
with the variable is replaced a new value.
The *action* may also be related to the current stored value in some way 
such as a counter for the number of visits. 

```javascript 
    const initialState = 0;
    export const variable = (state=initialState, action) => {
        switch (action.type) {
            case ACTION_NAME: {
                return state + action.body.variable;
            }
            default: {
                return state;
            }
        }
    };
```

This definition can be used as a counter starting from 0 and increasing 
with each call to **ACTION_NAME**.

## Storing Session Information 

To store session information, we must 

- import the `useDispatch` hook 
- import the action associated with the variable to update 
- define the `dispatch` variable 
- call the action to update the session information 

The following example refers to the actions defined in the previous section.

```javascript
    import {useDispatch} from 'react-redux';
    import {actionName} from '[file-path]/actions';

    const ReactElement = () => {
        const dispatch = useDispatch();
        ...
        { // Section to update variable
            dispatch(actionName(updateValue));
        }
        ...
    };
```

## Retrieving Session Information 

To retrieve session information, we must 

- import the `useSelector` hook

### Example 

```javascript
    import {useSelector} from 'react-redux';

    const ReactElement = () => {
        const variable = useSelector(store => store.variable);s
    };
```

## Other Information 

Additional information that may be useful to know.

- Redux is linked to the React application in the [index file](../../React/src/index.js).
- The [store file](../../react/src/redux/store.js) combines the functions and associated variable names together. (No need to make changes unless the filesystem changes. eg. moving files outside the redux file).
- Session storage must be manually wiped when logging out as otherwise the information may persist when logging out and in. 
This information wipe occurs in the [index reducer file](../../React/src/redux/reducers/index.js) where the overall state can be changed.

## Useful Tools

The following are useful tools:

- [Redux Devtools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd): Chrome extention tat shows the current state of the Redux store.

