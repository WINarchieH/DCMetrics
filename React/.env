REACT_APP_EXPRESS_URL='http://localhost:5000'

REACT_APP_SSRS_URL='http://dcm1/ReportServer/Pages/ReportViewer.aspx?'















                   
                  
             

    
              
 
                  

                  


 
                  






              