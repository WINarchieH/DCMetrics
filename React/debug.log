[1230/142438.358:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
[0111/093331.713:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
[0121/093816.694:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
[0128/124148.874:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
