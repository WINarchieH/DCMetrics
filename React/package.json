{
  "name": "web",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@devexpress/dx-react-chart-material-ui": "^2.7.1",
    "@emotion/react": "^11.4.1",
    "@emotion/styled": "^11.3.0",
    "@iconify/icons-ant-design": "^1.1.0",
    "@iconify/icons-eva": "^1.1.0",
    "@iconify/icons-ic": "^1.1.10",
    "@iconify/react": "^3.0.1",
    "@material-ui/core": "^4.11.0",
    "@material-ui/icons": "^4.9.1",
    "@material-ui/lab": "^4.0.0-alpha.58",
    "@material-ui/pickers": "^3.2.10",
    "@mui/icons-material": "^5.0.3",
    "@mui/material": "^5.0.4",
    "@mui/styles": "^5.0.1",
    "@reach/dialog": "^0.13.0",
    "@tabler/icons": "^1.41.2",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.5.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/date-fns": "^2.6.0",
    "@welldone-software/why-did-you-render": "^4.2.7",
    "ag-grid-community": "^25.3.0",
    "ag-grid-react": "^25.3.0",
    "apexcharts": "^3.35.0",
    "bootstrap": "^4.5.2",
    "canvasjs": "^1.8.3",
    "canvasjs-react-charts": "^1.0.5",
    "change-case": "^4.1.2",
    "chart.js": "^2.9.4",
    "csv-converter-to-pdf-and-html": "^1.0.2",
    "date-fns": "^2.0.0-alpha.27",
    "devextreme": "20.1.7",
    "devextreme-react": "20.1.7",
    "formik": "^2.1.7",
    "jspdf": "^2.3.1",
    "jspdf-autotable": "^3.5.13",
    "material-table": "^1.66.0",
    "material-ui-popup-state": "^2.0.0",
    "moment": "^2.29.1",
    "moment-timezone": "^0.5.33",
    "mui-rte": "^1.29.0",
    "node-sass": "^4.14.1",
    "papaparse": "^5.3.0",
    "react": "^16.14.0",
    "react-apexcharts": "^1.4.0",
    "react-app-polyfill": "^1.0.6",
    "react-autosize-textarea": "^7.1.0",
    "react-beautiful-dnd": "^13.0.0",
    "react-big-calendar": "^0.35.0",
    "react-bootstrap": "^1.3.0",
    "react-chartjs-2": "^2.10.0",
    "react-charts": "^2.0.0-beta.7",
    "react-checkbox-list": "0.0.2",
    "react-dates": "^21.8.0",
    "react-dom": "^16.14.0",
    "react-excel-renderer": "^1.1.0",
    "react-file-drop": "^3.1.2",
    "react-gif-player": "^0.4.2",
    "react-google-charts": "^3.0.15",
    "react-helmet": "^6.1.0",
    "react-linechart": "^1.1.12",
    "react-moment": "^1.1.1",
    "react-multi-select-component": "^4.3.4",
    "react-multiselect-checkboxes": "^0.1.1",
    "react-notification-badge": "^1.5.1",
    "react-notifications-component": "^3.1.0",
    "react-pdf": "^5.3.2",
    "react-perfect-scrollbar": "^1.5.8",
    "react-redux": "^7.2.0",
    "react-responsive-carousel": "^3.2.10",
    "react-router-dom": "^5.1.2",
    "react-scripts": "^3.4.3",
    "react-select": "^3.1.0",
    "react-spinners": "^0.11.0",
    "react-table": "^7.5.0",
    "react-table-plugins": "^1.3.1",
    "react-toastify": "^8.0.3",
    "reactstrap": "^8.5.1",
    "redux": "^4.0.5",
    "save": "^2.4.0",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.0.3",
    "simplebar-react": "^2.3.6",
    "yargs-parser": "^18.1.3",
    "yup": "^0.29.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version",
      "ie >= 9"
    ]
  },
  "devDependencies": {
    "@iconify/react": "^3.0.1",
    "axios": "^0.21.1",
    "faker": "^4.1.0",
    "file-saver": "^2.0.5",
    "puppeteer": "^5.2.1",
    "react-test-renderer": "^16.13.1",
    "xlsx": "^0.17.1"
  }
}
