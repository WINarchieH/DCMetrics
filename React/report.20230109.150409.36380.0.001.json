
{
  "header": {
    "reportVersion": 2,
    "event": "Allocation failed - JavaScript heap out of memory",
    "trigger": "FatalError",
    "filename": "report.20230109.150409.36380.0.001.json",
    "dumpEventTime": "2023-01-09T15:04:09Z",
    "dumpEventTimeStamp": "1673237049749",
    "processId": 36380,
    "threadId": null,
    "cwd": "C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React",
    "commandLine": [
      "node",
      "C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React\\node_modules\\react-scripts\\scripts\\build.js"
    ],
    "nodejsVersion": "v12.16.3",
    "wordSize": 64,
    "arch": "x64",
    "platform": "win32",
    "componentVersions": {
      "node": "12.16.3",
      "v8": "7.8.279.23-node.35",
      "uv": "1.34.2",
      "zlib": "1.2.11",
      "brotli": "1.0.7",
      "ares": "1.16.0",
      "modules": "72",
      "nghttp2": "1.40.0",
      "napi": "5",
      "llhttp": "2.0.4",
      "http_parser": "2.9.3",
      "openssl": "1.1.1g",
      "cldr": "36.0",
      "icu": "65.1",
      "tz": "2019c",
      "unicode": "12.1"
    },
    "release": {
      "name": "node",
      "lts": "Erbium",
      "headersUrl": "https://nodejs.org/download/release/v12.16.3/node-v12.16.3-headers.tar.gz",
      "sourceUrl": "https://nodejs.org/download/release/v12.16.3/node-v12.16.3.tar.gz",
      "libUrl": "https://nodejs.org/download/release/v12.16.3/win-x64/node.lib"
    },
    "osName": "Windows_NT",
    "osRelease": "10.0.22000",
    "osVersion": "Windows 10 Home",
    "osMachine": "x86_64",
    "cpus": [
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 16798265,
        "nice": 0,
        "sys": 20658343,
        "idle": 215686125,
        "irq": 2701578
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 21485390,
        "nice": 0,
        "sys": 13083828,
        "idle": 218573406,
        "irq": 341187
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 25310140,
        "nice": 0,
        "sys": 18413687,
        "idle": 209418781,
        "irq": 588671
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 18577171,
        "nice": 0,
        "sys": 15869406,
        "idle": 218696031,
        "irq": 640968
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 23006375,
        "nice": 0,
        "sys": 18000656,
        "idle": 212135578,
        "irq": 662640
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 18455343,
        "nice": 0,
        "sys": 13979015,
        "idle": 220708218,
        "irq": 426312
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 22153921,
        "nice": 0,
        "sys": 18029468,
        "idle": 212959203,
        "irq": 704937
      },
      {
        "model": "Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz",
        "speed": 2304,
        "user": 19804406,
        "nice": 0,
        "sys": 14638843,
        "idle": 218699343,
        "irq": 382421
      }
    ],
    "networkInterfaces": [
      {
        "name": "Wi-Fi",
        "internal": false,
        "mac": "c8:09:a8:7c:6b:22",
        "address": "fe80::1cb:c0e4:903:1c07",
        "netmask": "ffff:ffff:ffff:ffff::",
        "family": "IPv6",
        "scopeid": 13
      },
      {
        "name": "Wi-Fi",
        "internal": false,
        "mac": "c8:09:a8:7c:6b:22",
        "address": "192.168.3.105",
        "netmask": "255.255.255.0",
        "family": "IPv4"
      },
      {
        "name": "Loopback Pseudo-Interface 1",
        "internal": true,
        "mac": "00:00:00:00:00:00",
        "address": "::1",
        "netmask": "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff",
        "family": "IPv6",
        "scopeid": 0
      },
      {
        "name": "Loopback Pseudo-Interface 1",
        "internal": true,
        "mac": "00:00:00:00:00:00",
        "address": "127.0.0.1",
        "netmask": "255.0.0.0",
        "family": "IPv4"
      }
    ],
    "host": "DESKTOP-JM8BRCG"
  },
  "javascriptStack": {
    "message": "No stack.",
    "stack": [
      "Unavailable."
    ]
  },
  "nativeStack": [
    {
      "pc": "0x00007ff762462449",
      "symbol": "std::basic_ostream<char,std::char_traits<char> >::operator<<+11577"
    },
    {
      "pc": "0x00007ff7624669a9",
      "symbol": "std::basic_ostream<char,std::char_traits<char> >::operator<<+29337"
    },
    {
      "pc": "0x00007ff762465828",
      "symbol": "std::basic_ostream<char,std::char_traits<char> >::operator<<+24856"
    },
    {
      "pc": "0x00007ff76259d8c2",
      "symbol": "v8::base::CPU::has_sse+68658"
    },
    {
      "pc": "0x00007ff762db9bbe",
      "symbol": "v8::Isolate::ReportExternalAllocationLimitReached+94"
    },
    {
      "pc": "0x00007ff762da1c91",
      "symbol": "v8::SharedArrayBuffer::Externalize+833"
    },
    {
      "pc": "0x00007ff762c6e1ec",
      "symbol": "v8::internal::Heap::EphemeronKeyWriteBarrierFromCode+1436"
    },
    {
      "pc": "0x00007ff762c79420",
      "symbol": "v8::internal::Heap::ProtectUnprotectedMemoryChunks+1312"
    },
    {
      "pc": "0x00007ff762c75f44",
      "symbol": "v8::internal::Heap::PageFlagsAreConsistent+3204"
    },
    {
      "pc": "0x00007ff762c6b743",
      "symbol": "v8::internal::Heap::CollectGarbage+1283"
    },
    {
      "pc": "0x00007ff762c69db4",
      "symbol": "v8::internal::Heap::AddRetainedMap+2452"
    },
    {
      "pc": "0x00007ff762c8afbd",
      "symbol": "v8::internal::Factory::NewFillerObject+61"
    },
    {
      "pc": "0x00007ff7629f1871",
      "symbol": "v8::internal::interpreter::JumpTableTargetOffsets::iterator::operator=+1665"
    },
    {
      "pc": "0x00007ff763206c4d",
      "symbol": "v8::internal::SetupIsolateDelegate::SetupHeap+546637"
    },
    {
      "pc": "0x00007ff76318d0f1",
      "symbol": "v8::internal::SetupIsolateDelegate::SetupHeap+48113"
    },
    {
      "pc": "0x00000157bcbb8f55",
      "symbol": ""
    }
  ],
  "javascriptHeap": {
    "totalMemory": 2183327744,
    "totalCommittedMemory": 2183327744,
    "usedMemory": 2133998848,
    "availableMemory": 53463856,
    "memoryLimit": 2197815296,
    "heapSpaces": {
      "read_only_space": {
        "memorySize": 262144,
        "committedMemory": 262144,
        "capacity": 32808,
        "used": 32808,
        "available": 0
      },
      "new_space": {
        "memorySize": 33554432,
        "committedMemory": 33554432,
        "capacity": 16759296,
        "used": 3039240,
        "available": 13720056
      },
      "old_space": {
        "memorySize": 1612853248,
        "committedMemory": 1612853248,
        "capacity": 1604878952,
        "used": 1596382000,
        "available": 8496952
      },
      "code_space": {
        "memorySize": 2527232,
        "committedMemory": 2527232,
        "capacity": 2117216,
        "used": 2117216,
        "available": 0
      },
      "map_space": {
        "memorySize": 5771264,
        "committedMemory": 5771264,
        "capacity": 4474640,
        "used": 4474640,
        "available": 0
      },
      "large_object_space": {
        "memorySize": 527736832,
        "committedMemory": 527736832,
        "capacity": 527408656,
        "used": 527408656,
        "available": 0
      },
      "code_large_object_space": {
        "memorySize": 622592,
        "committedMemory": 622592,
        "capacity": 544288,
        "used": 544288,
        "available": 0
      },
      "new_large_object_space": {
        "memorySize": 0,
        "committedMemory": 0,
        "capacity": 16759296,
        "used": 0,
        "available": 16759296
      }
    }
  },
  "resourceUsage": {
    "userCpuSeconds": 183.046,
    "kernelCpuSeconds": 17.125,
    "cpuConsumptionPercent": 106.474,
    "maxRss": 2445144064,
    "pageFaults": {
      "IORequired": 2180112,
      "IONotRequired": 0
    },
    "fsActivity": {
      "reads": 15078,
      "writes": 47
    }
  },
  "libuv": [
  ],
  "workers": [
  ],
  "environmentVariables": {
    "ALLUSERSPROFILE": "C:\\ProgramData",
    "APPDATA": "C:\\Users\\61405\\AppData\\Roaming",
    "asl.log": "Destination=file",
    "BABEL_ENV": "production",
    "COLOR": "1",
    "CommonProgramFiles": "C:\\Program Files\\Common Files",
    "CommonProgramFiles(x86)": "C:\\Program Files (x86)\\Common Files",
    "CommonProgramW6432": "C:\\Program Files\\Common Files",
    "COMPUTERNAME": "DESKTOP-JM8BRCG",
    "ComSpec": "C:\\WINDOWS\\system32\\cmd.exe",
    "DriverData": "C:\\Windows\\System32\\Drivers\\DriverData",
    "EDITOR": "notepad.exe",
    "FPS_BROWSER_APP_PROFILE_STRING": "Internet Explorer",
    "FPS_BROWSER_USER_PROFILE_STRING": "Default",
    "HOME": "C:\\Users\\61405",
    "HOMEDRIVE": "C:",
    "HOMEPATH": "\\Users\\61405",
    "INIT_CWD": "C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React",
    "LOCALAPPDATA": "C:\\Users\\61405\\AppData\\Local",
    "LOGONSERVER": "\\\\DESKTOP-JM8BRCG",
    "MSMPI_BIN": "C:\\Program Files\\Microsoft MPI\\Bin\\",
    "NODE": "C:\\Program Files\\nodejs\\node.exe",
    "NODE_ENV": "production",
    "NODE_EXE": "C:\\Program Files\\nodejs\\\\node.exe",
    "NODE_PATH": "",
    "NPM_CLI_JS": "C:\\Users\\61405\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js",
    "npm_command": "run-script",
    "npm_config_cache": "C:\\Users\\61405\\AppData\\Local\\npm-cache",
    "npm_config_globalconfig": "C:\\Users\\61405\\AppData\\Roaming\\npm\\etc\\npmrc",
    "npm_config_init_module": "C:\\Users\\61405\\.npm-init.js",
    "npm_config_metrics_registry": "https://registry.npmjs.org/",
    "npm_config_node_gyp": "C:\\Users\\61405\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js",
    "npm_config_noproxy": "",
    "npm_config_prefix": "C:\\Users\\61405\\AppData\\Roaming\\npm",
    "npm_config_userconfig": "C:\\Users\\61405\\.npmrc",
    "npm_config_user_agent": "npm/7.19.0 node/v12.16.3 win32 x64 workspaces/false",
    "npm_execpath": "C:\\Users\\61405\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js",
    "npm_lifecycle_event": "build",
    "npm_lifecycle_script": "react-scripts build",
    "npm_node_execpath": "C:\\Program Files\\nodejs\\node.exe",
    "npm_package_json": "C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React\\package.json",
    "npm_package_name": "web",
    "npm_package_version": "0.1.0",
    "NPM_PREFIX_NPM_CLI_JS": "C:\\Users\\61405\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js",
    "NUMBER_OF_PROCESSORS": "8",
    "OneDrive": "C:\\Users\\61405\\OneDrive",
    "OneDriveConsumer": "C:\\Users\\61405\\OneDrive",
    "OS": "Windows_NT",
    "Path": "C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React\\node_modules\\.bin;C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\node_modules\\.bin;C:\\Shitiz\\clients\\Makita\\DCM\\node_modules\\.bin;C:\\Shitiz\\clients\\Makita\\node_modules\\.bin;C:\\Shitiz\\clients\\node_modules\\.bin;C:\\Shitiz\\node_modules\\.bin;C:\\node_modules\\.bin;C:\\Users\\61405\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\@npmcli\\run-script\\lib\\node-gyp-bin;C:\\Program Files\\Microsoft MPI\\Bin\\;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\dotnet\\;C:\\Program Files\\Microsoft SQL Server\\130\\Tools\\Binn\\;C:\\Program Files\\Microsoft SQL Server\\Client SDK\\ODBC\\170\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\150\\Tools\\Binn\\;C:\\Program Files\\Microsoft SQL Server\\150\\Tools\\Binn\\;C:\\Program Files\\Microsoft SQL Server\\150\\DTS\\Binn\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\nodejs\\;C:\\Program Files\\Microsoft\\Web Platform Installer\\;C:\\Program Files (x86)\\Microsoft SQL Server\\Client SDK\\ODBC\\130\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\DTS\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\ManagementStudio\\;C:\\Program Files\\Microsoft SQL Server\\Client SDK\\ODBC\\130\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\130\\Tools\\Binn\\;C:\\Program Files\\Microsoft SQL Server\\130\\DTS\\Binn\\;C:\\Program Files\\TortoiseGit\\bin;C:\\Program Files\\MySQL\\MySQL Shell 8.0\\bin\\;C:\\Users\\61405\\AppData\\Roaming\\npm;C:\\Users\\61405\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\61405\\AppData\\Local\\Microsoft\\WindowsApps;",
    "PATHEXT": ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC",
    "PROCESSOR_ARCHITECTURE": "AMD64",
    "PROCESSOR_IDENTIFIER": "Intel64 Family 6 Model 142 Stepping 12, GenuineIntel",
    "PROCESSOR_LEVEL": "6",
    "PROCESSOR_REVISION": "8e0c",
    "ProgramData": "C:\\ProgramData",
    "ProgramFiles": "C:\\Program Files",
    "ProgramFiles(x86)": "C:\\Program Files (x86)",
    "ProgramW6432": "C:\\Program Files",
    "PROMPT": "$P$G",
    "PSModulePath": "C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules;C:\\Program Files (x86)\\Microsoft SQL Server\\150\\Tools\\PowerShell\\Modules\\;C:\\Program Files (x86)\\Microsoft SQL Server\\130\\Tools\\PowerShell\\Modules\\",
    "PUBLIC": "C:\\Users\\Public",
    "REACT_APP_EXPRESS_URL": "http://192.168.10.249:91",
    "REACT_APP_SSRS_URL": "http://dcmetrics/ReportServer/Pages/ReportViewer.aspx?",
    "SESSIONNAME": "Console",
    "SystemDrive": "C:",
    "SystemRoot": "C:\\WINDOWS",
    "TEMP": "C:\\Users\\61405\\AppData\\Local\\Temp",
    "TMP": "C:\\Users\\61405\\AppData\\Local\\Temp",
    "USERDOMAIN": "DESKTOP-JM8BRCG",
    "USERDOMAIN_ROAMINGPROFILE": "DESKTOP-JM8BRCG",
    "USERNAME": "61405",
    "USERPROFILE": "C:\\Users\\61405",
    "windir": "C:\\WINDOWS",
    "ZES_ENABLE_SYSMAN": "1",
    "_CSRUN_DISABLE_WORKAROUNDS": "1",
    "__PSLockDownPolicy": "0"
  },
  "sharedObjects": [
    "C:\\Program Files\\nodejs\\node.exe",
    "C:\\WINDOWS\\SYSTEM32\\ntdll.dll",
    "C:\\WINDOWS\\System32\\KERNEL32.dll",
    "C:\\Windows\\system32\\hmpalert.dll",
    "C:\\WINDOWS\\System32\\KERNELBASE.dll",
    "C:\\WINDOWS\\System32\\WS2_32.dll",
    "C:\\WINDOWS\\System32\\RPCRT4.dll",
    "C:\\WINDOWS\\System32\\PSAPI.DLL",
    "C:\\WINDOWS\\System32\\ADVAPI32.dll",
    "C:\\WINDOWS\\System32\\msvcrt.dll",
    "C:\\WINDOWS\\System32\\sechost.dll",
    "C:\\WINDOWS\\System32\\USER32.dll",
    "C:\\WINDOWS\\System32\\win32u.dll",
    "C:\\WINDOWS\\System32\\GDI32.dll",
    "C:\\WINDOWS\\System32\\gdi32full.dll",
    "C:\\WINDOWS\\System32\\msvcp_win.dll",
    "C:\\WINDOWS\\System32\\ucrtbase.dll",
    "C:\\WINDOWS\\System32\\CRYPT32.dll",
    "C:\\WINDOWS\\SYSTEM32\\dbghelp.dll",
    "C:\\WINDOWS\\SYSTEM32\\IPHLPAPI.DLL",
    "C:\\WINDOWS\\SYSTEM32\\USERENV.dll",
    "C:\\WINDOWS\\SYSTEM32\\bcrypt.dll",
    "C:\\WINDOWS\\SYSTEM32\\WINMM.dll",
    "C:\\WINDOWS\\System32\\IMM32.DLL",
    "C:\\WINDOWS\\SYSTEM32\\powrprof.dll",
    "C:\\WINDOWS\\SYSTEM32\\UMPDC.dll",
    "C:\\WINDOWS\\SYSTEM32\\CRYPTBASE.DLL",
    "C:\\WINDOWS\\system32\\uxtheme.dll",
    "C:\\WINDOWS\\System32\\combase.dll",
    "C:\\WINDOWS\\system32\\mswsock.dll",
    "C:\\WINDOWS\\SYSTEM32\\kernel.appcore.dll",
    "C:\\WINDOWS\\System32\\bcryptprimitives.dll",
    "C:\\WINDOWS\\SYSTEM32\\ntmarta.dll",
    "\\\\?\\C:\\Shitiz\\clients\\Makita\\DCM\\DCMetrics\\React\\node_modules\\node-sass\\vendor\\win32-x64-72\\binding.node",
    "C:\\WINDOWS\\System32\\NSI.dll",
    "C:\\WINDOWS\\SYSTEM32\\dhcpcsvc6.DLL",
    "C:\\WINDOWS\\SYSTEM32\\dhcpcsvc.DLL",
    "C:\\WINDOWS\\SYSTEM32\\DNSAPI.dll",
    "C:\\WINDOWS\\system32\\napinsp.dll",
    "C:\\WINDOWS\\system32\\pnrpnsp.dll",
    "C:\\WINDOWS\\System32\\winrnr.dll",
    "C:\\WINDOWS\\system32\\wshbth.dll",
    "C:\\WINDOWS\\system32\\nlansp_c.dll",
    "C:\\Program Files\\Bonjour\\mdnsNSP.dll"
  ]
}