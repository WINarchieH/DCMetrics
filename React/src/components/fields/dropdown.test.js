import React from 'react';
import renderer from 'react-test-renderer';
import DropDown from './dropdown';

it('Dropdown Renders', () => {
    const tree = renderer
        .create(<DropDown label='test' onchange={(x) => x}
            options={[]}></DropDown>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Dropdown with Inputs', () => {
    const tree = renderer
        .create(<DropDown label='Gender' onchange={(x) => x}
            options={['M', 'F', 'X']} optionNames={['Male', 'Female', 'Unspecified']}
            required
        ></DropDown>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

// If snapshot changed intentionally
// jest --updateSnapshot 
// https://jestjs.io/docs/en/snapshot-testing