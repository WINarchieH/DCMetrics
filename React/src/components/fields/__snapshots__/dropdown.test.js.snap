// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Renders 1`] = `
<div
  className="modal-item"
>
  <label
    className="label label--position"
  >
    test
  </label>
  <select
    className="modal-fields modal-fields--outline"
    onChange={[Function]}
    value=""
  >
    <option
      className="dropdown-item"
      value=""
    >
      Select 
      test
    </option>
  </select>
</div>
`;

exports[`Dropdown with Inputs 1`] = `
<div
  className="modal-item"
>
  <label
    className="label label--position"
  >
    Gender
  </label>
  <select
    className="modal-fields modal-fields--outline"
    onChange={[Function]}
    required={true}
    value=""
  >
    <option
      className="dropdown-item"
      value=""
    >
      Select 
      gender
    </option>
    <option
      className="dropdown-item"
      value="M"
    >
      Male
    </option>
    <option
      className="dropdown-item"
      value="F"
    >
      Female
    </option>
    <option
      className="dropdown-item"
      value="X"
    >
      Unspecified
    </option>
  </select>
</div>
`;
